SIMS/SIMS/Inventory.cs

```diff
@@ -0,0 +1,122 @@
+namespace SIMS;
+
+public class Inventory
+{
+    // Products property
+    private List<Product> Products { get; set; } = [];
+
+    // AddProduct method
+    public bool AddProduct(Product? product)
+    {
+        //check if the product is null
+        if (product is null)
+        {
+            return false;
+        }
+
+        //check if the product already exists
+        if (Products.Any(p => p.Name == product.Name))
+        {
+            return false;
```

Comment:
how would the client (the main function in this case) distinguish between this case and the previous one when the product is null? what about throwing exceptions with meaningful err msgs?